wire-java-generator/src/main/java/com/squareup/wire/java/JavaGenerator.java

```diff
@@ -423,7 +423,22 @@ private static void putAllExtensions(
         if (!eligibleAsAnnotationMember(schema, field)) continue;
 
         ProtoMember protoMember = extend.member(field);
-        memberToJavaName.put(protoMember, annotationName(protoFile, field, new ClassNameFactory()));
+        ClassName annotationName =
+            annotationName(protoFile, field, new ClassNameFactory(), "Option");
+        if (memberToJavaName.containsValue(annotationName)) {
+          // To avoid conflicts for same named options on different types, we generate a more
```

Comment:
Is it possible for us to always pick suffix based on the option type, so that all message options will be named `XxxMessageOption` and all field names - `XxxFieldOption`?

Reply:
That'd be the ideal solution but the feature has already shipped and we'll be breaking all clients with this change 😢 
So we decided to do that only when conflicts arise.